content/asm_6.md

```diff
@@ -1,181 +1,561 @@
+
+### Data transfer instructions
+
+As we know from the [Data manipulation](./asm_4.md) chapter, data transfer instructions are used to move data between different locations. The `x86_64` CPU provides a special set of instructions to transfer floating-point data. If you are going to use the `x87 FPU` unit, the most common data transfer instructions are:
+
+- `fld` - Loads a floating point value.
+- `fst` - Stores a floating point value.
+
+These instructions are used by the `x87 FPU` unit to store and load the floating-point numbers at/from the stack.
+
+To work with the `XMM` registers, the `x86_64` CPU supports, among others, the following data transfer instructions:
+
+- `movss` - Moves a single-precision floating-point value between the `XMM` registers or between an `XMM` register and memory.
+- `movsd` - Moves double-precision floating-point value between the `XMM` registers or between an `XMM` register and memory.
+- `movhlps` - Moves two-packed single-precision floating-point values from the high quadword of an `XMM` register to the low quadword of another `XMM` register.
+- `movlhps` - Moves two-packed single-precision floating-point values from the low quadword of an `XMM` register to the high quadword of another `XMM` register.
+
+### Floating-point arithmetic instructions
+
+The floating-point arithmetic instructions perform arithmetic operations, such as addition, subtraction, multiplication, and division, on single or double-precision floating-point values. The following floating-point arithmetic instructions exist:
+
+- `addss` - Adds a single-precision floating-point values.
+- `addsd` - Adds a double-precision floating point values.
+- `subss` - Subtracts a single-precision floating-point values.
+- `subsd` - Subtracts a double-precision floating-point values.
+- `mulss` - Multiplies a single-precision floating-point values.
+- `mulsd` - Multiplies a double-precision floating-point values.
+- `divss` - Divides a single-precision floating-point values.
+- `divsd` - Divides a double-precision floating-point values.
+
+All of these instructions expect two operands to execute the given operation of addition, subtraction, multiplication, or division. After the operation is executed, the result is stored in the first operand.
+
+### Floating-Point control instructions
+
+As we already know, the main goal of these instructions is to manage the control flow of our programs. For integer comparisons, we used the `cmp` instruction. But this instruction does not work with floating-point values, even though the result of the comparison is controlled by the same `rflags` registers that we saw in the [Data manipulation](./asm_4.md#control-transfer-instructions) chapter. The general instruction to compare floating-point numbers is `cmpss`. Instead of setting the value of the flag based on the result comparison, the instruction stores the result of the comparison in the destination register.
```

Comment:
```suggestion
As we already know, the main goal of these instructions is to manage the control flow of our programs. For integer comparisons, we used the `cmp` instruction. However, this instruction does not work with floating-point values, even though the result of the comparison is controlled by the same `rflags` registers that we saw in the [Data manipulation](./asm_4.md#control-transfer-instructions) chapter. The general instruction to compare floating-point numbers is `cmpss`. Instead of setting the value of the flag based on the result comparison, the instruction stores the result of the comparison in the destination register.
```